### DISCUSSION TITLE
Restricting generated queries to only the signed in users data

### DISCUSSION THREAD

**Discussion by ghost** [MichalLytek/typegraphql-prisma#299] [Q&A] [answered]
Hello,

I am currently trying to restrict the data coming back from the generated Query Resolvers to only be the data of a signed in user.

For example:

```ts

model User {
    id        String          @id @unique @default(cuid())
    ...
    posts Post[]
}

model Post {
    id        String          @id @unique @default(cuid())
    ...
    user User
}
```

User Tom is signed in and he should only see his posts. Instead of passing the following query:

```gql
query posts {
  posts(input: { where: { userId: <Tom's ID>, isPublished: false} }) {
    id
  }
}
```
to directly query his posts, you simply do the following

```gql
query posts {
  posts(input: { where: { isPublished: false} }) {
    id 
  }
}
```

and it will only query the posts created by Tom

You could achieve this easily by creating your own CustomResolver and simply pulling the UserId from the `@Ctx()` but I would like to keep all the prebuilt filtering!

Is this even possible currently? If so, would the best way to achieve this be writing a custom decorator/middleware function that you apply to every generated query via the `applyResolversEnhanceMap`?

Thanks,

**Comment by ghost** [comment]
What I am currently doing is just creating a custom resolver that leverages some of the generated types and helper functions. This seems to be working for now but not sure if there is a better solution.

```ts
  @Authorized()
  @Query(() => [Post])
  async posts(
    @Ctx() { prisma, user }: Context,
    @Info() info: GraphQLResolveInfo,
    @Args() args: FindManyPostArgs
  ) {
    const { _count } = transformFields(graphqlFields(info as any));
    const overrideWhere: PostWhereInput = {
      ...args.where,
      userId: { equals: user.id },
    };
    const newArgs = { ...args, where: overrideWhere };
    return await prisma.post.findMany({
      ...newArgs,
      ...(_count && transformCountFieldIntoSelectRelationsCount(_count)),
    });
  }
```

**Comment by MichalLytek** [comment, marked as answer]
If you don't want to reimplement the resolver method body, you can try to extend the generated class and pass there the modified arguments.
